pkg/config/config.go

```diff
@@ -211,23 +213,51 @@ func (c *MicroshiftConfig) updateManifestList() {
 }
 
 func (c *MicroshiftConfig) ReadFromCmdLine(flags *pflag.FlagSet) error {
-	if dataDir, err := flags.GetString("data-dir"); err == nil && flags.Changed("data-dir") {
-		c.DataDir = dataDir
+	if s, err := flags.GetString("data-dir"); err == nil && flags.Changed("data-dir") {
+		c.DataDir = s
 		// if the defaults are present, rebuild based on the new data-dir
 		c.updateManifestList()
 	}
-	if auditLogDir, err := flags.GetString("audit-log-dir"); err == nil && flags.Changed("audit-log-dir") {
-		c.AuditLogDir = auditLogDir
+	if s, err := flags.GetString("audit-log-dir"); err == nil && flags.Changed("audit-log-dir") {
+		c.AuditLogDir = s
 	}
-	if vLevelFlag := flags.Lookup("v"); vLevelFlag != nil && flags.Changed("v") {
-		c.LogVLevel, _ = strconv.Atoi(vLevelFlag.Value.String())
+	if f := flags.Lookup("v"); f != nil && flags.Changed("v") {
+		c.LogVLevel, _ = strconv.Atoi(f.Value.String())
 	}
-	if roles, err := flags.GetStringSlice("roles"); err == nil && flags.Changed("roles") {
-		c.Roles = roles
+	if ss, err := flags.GetStringSlice("roles"); err == nil && flags.Changed("roles") {
+		c.Roles = ss
 	}
-	if pprofFlag, err := flags.GetBool("debug.pprof"); err == nil && flags.Changed("debug.pprof") {
-		c.Debug.Pprof = pprofFlag
+	if s, err := flags.GetString("node-name"); err == nil && flags.Changed("node-name") {
+		c.NodeName = s
 	}
+	if s, err := flags.GetString("node-ip"); err == nil && flags.Changed("node-ip") {
```

Comment:
I wouldn't expect to need to do all of this work to copy options one at a time if the flags are bound to the config struct by addRunFlags. Maybe I don't understand how that works?